whois-api/src/main/java/net/ripe/db/whois/api/rest/ClientCertificateExtractor.java

```diff
@@ -0,0 +1,65 @@
+package net.ripe.db.whois.api.rest;
+
+import net.ripe.db.whois.common.io.ByteArrayInput;
+import org.apache.commons.lang.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.servlet.http.HttpServletRequest;
+import java.security.Principal;
+import java.security.cert.CertificateException;
+import java.security.cert.CertificateExpiredException;
+import java.security.cert.CertificateFactory;
+import java.security.cert.CertificateNotYetValidException;
+import java.security.cert.X509Certificate;
+import java.util.Optional;
+
+import static org.apache.commons.codec.binary.Base64.decodeBase64;
+
+public class ClientCertificateExtractor {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ClientCertificateExtractor.class);
+
+    private final static String HEADER_SSL_CLIENT_CERT = "SSL_CLIENT_CERT";
+    private final static String HEADER_SSL_CLIENT_VERIFY = "SSL_CLIENT_VERIFY";
+
+    public static Optional<X509Certificate> getClientCertificate(final HttpServletRequest request) {
+        final String sslClientCert = request.getHeader(HEADER_SSL_CLIENT_CERT);
+
+        if (StringUtils.isBlank(sslClientCert)) {
+            return Optional.empty();
+        }
+
+        final String sslClientVerify = request.getHeader(HEADER_SSL_CLIENT_VERIFY);
```

Comment:
This header doesn't mean very much, as the proxy doesn't check the CA. Do we need to include it?

Reply:
As specified in the requirements, we should check the value is as expected.,

Reply:
We're expecting Apache to do the verification and tell us:

SSL_CLIENT_VERIFY | string | NONE, SUCCESS, GENEROUS or FAILED:reason
-- | -- | --

If it's not GENEROUS or SUCCESS we don't accept the cert and otherwise we do but we check it again.